tools/docker-images/clp-package/build.sh

```diff
@@ -3,17 +3,38 @@
 set -eu
 set -o pipefail
 
+remove_temp_file_and_prev_image() {
+    rm -f "$temp_iid_file"
+
+    [[ -z "$prev_image_id" || "$prev_image_id" == "$new_image_id" ]] && return
+
+    if docker image inspect "$prev_image_id" >/dev/null 2>&1; then
+        echo "Removing previous image $prev_image_id."
+        docker image remove "$prev_image_id"
```

Comment:
The image might fail to be removed because some running container might still be referencing it, which causes a non-zero exit code in the script.

When the image fails to be removed, we should either exit gracefully (with 0) with a warning message printed, or the caller should gracefully handle the non-exit code. Currently, the caller (the docker-images.yaml taskfile) doesn't handle this --- when the script fails, any commands following the script call (e.g., to copy the unique image id file into `build/clp-package`) would fail to execute.

@coderabbitai create an issue to track this